src/components/GridContainer/GridContainer.tsx

```diff
@@ -0,0 +1,129 @@
+export type FlowOptions = "row" | "column" | "row-dense" | "column-dense";
+type GapOptions = "none" | "xxs" | "xs" | "sm" | "md" | "lg" | "xl" | "xxl" | "unset";
+type ItemsOptions = "start" | "center" | "end" | "stretch";
+type ContentOptions =
+  | "center"
+  | "space-between"
+  | "space-around"
+  | "space-evenly"
+  | "start"
+  | "stretch"
+  | "end"
+  | "left"
+  | "right";
+
+export interface GridContainerProps extends HTMLAttributes<HTMLDivElement> {
+  alignItems?: ItemsOptions;
+  alignContent?: ContentOptions;
+  children?: React.ReactNode;
+  display?: DisplayOptions;
+  columnGap?: GapOptions;
+  gap?: GapOptions;
+  gridAutoColumns?: string;
+  gridAutoFlow?: FlowOptions;
+  gridAutoRows?: string;
+  gridTemplateAreas?: string;
+  gridTemplateColumns?: string;
+  gridTemplateRows?: string;
+  gridTemplate?: string;
+  isResponsive?: boolean;
+  justifyContent?: ContentOptions;
+  justifyItems?: ItemsOptions;
+  rowGap?: GapOptions;
+}
+
+const GridContainer = ({
+  alignItems = "stretch",
+  alignContent = "stretch",
+  children,
+  columnGap = "none",
+  display = "grid",
+  gap = "unset",
+  gridAutoColumns = "",
+  gridAutoFlow = "row",
+  gridAutoRows = "",
+  gridTemplateAreas = "",
+  gridTemplateColumns = "",
+  gridTemplateRows = "",
+  gridTemplate = "",
+  isResponsive = true,
+  justifyContent = "stretch",
+  justifyItems = "stretch",
+  rowGap = "none",
+  ...props
+}: GridContainerProps) => {
+  return (
+    <Wrapper
+      $alignItems={alignItems}
+      $alignContent={alignContent}
+      $columnGap={columnGap}
+      $display={display}
+      $gap={gap}
+      $gridAutoColumns={gridAutoColumns}
+      $gridAutoFlow={gridAutoFlow}
+      $gridAutoRows={gridAutoRows}
+      $gridTemplateAreas={gridTemplateAreas}
+      $gridTemplateColumns={gridTemplateColumns}
+      $gridTemplateRows={gridTemplateRows}
+      $gridTemplate={gridTemplate}
+      $isResponsive={isResponsive}
+      $justifyContent={justifyContent}
+      $justifyItems={justifyItems}
+      $rowGap={rowGap}
+      data-testid="grid-container"
+      {...props}
+    >
+      {children}
+    </Wrapper>
+  );
+};
+
+const Wrapper = styled.div<{
+  $alignContent: ContentOptions;
+  $alignItems: ItemsOptions;
+  $columnGap: GapOptions;
+  $display: DisplayOptions;
+  $gap: GapOptions;
+  $gridAutoColumns: string;
+  $gridAutoFlow: FlowOptions;
+  $gridAutoRows: string;
+  $gridTemplateAreas: string;
+  $gridTemplateColumns: string;
+  $gridTemplateRows: string;
+  $gridTemplate: string;
+  $isResponsive: boolean;
+  $justifyContent: ContentOptions;
+  $justifyItems: ItemsOptions;
+  $rowGap: GapOptions;
+}>`
+  align-items: ${({ $alignItems = "stretch" }) => $alignItems};
+  align-content: ${({ $alignContent = "stretch" }) => $alignContent};
+  column-gap: ${({ theme, $columnGap }) => theme.click.gridContainer.gap[$columnGap]};
+  display: ${({ $display }) => $display};
+  gap: ${({ theme, $gap }) => theme.click.gridContainer.gap[$gap]};
```

Comment:
Why are we declaring column-gap, row-gap and gap here?
We can use gap for column and row-gap
if column and row gap need to be overwritten then we should put it in the end and also in a conditional

Reply:
Yeah I've been playing around with this a little bit. At one point I thought about removing `gap` altogether as it makes things a lot more complicated. For now I opted against a conditional so it's relying on the user not to set both column-gap and gap.